File: app/src/main/java/co/smartreceipts/android/model/impl/columns/distance/DistanceLocationColumn.java
package co.smartreceipts.android.model.impl.columns.distance;

import android.content.Context;
import android.support.annotation.NonNull;

import java.util.List;

import co.smartreceipts.android.R;
import co.smartreceipts.android.model.Distance;
import co.smartreceipts.android.model.impl.columns.AbstractColumnImpl;
import co.smartreceipts.android.sync.model.SyncState;

public final class DistanceLocationColumn extends AbstractColumnImpl<Distance> {

    private final Context localizedContext;

    public DistanceLocationColumn(int id, @NonNull SyncState syncState, @NonNull Context context) {
        super(id, DistanceColumnDefinitions.ActualDefinition.LOCATION, syncState);
        localizedContext = context;
    }

    @Override
    public String getValue(@NonNull Distance distance) {
        return distance.getLocation();
    }

    @Override
    @NonNull
    public String getFooter(@NonNull List<Distance> distances) {
        return localizedContext.getString(R.string.total);
    }
}

File: app/src/main/kotlin/co/smartreceipts/android/model/impl/columns/distance/DistanceLocationColumn.kt
package co.smartreceipts.android.model.impl.columns.distance

import android.content.Context

import co.smartreceipts.android.R
import co.smartreceipts.android.model.Distance
import co.smartreceipts.android.model.impl.columns.AbstractColumnImpl
import co.smartreceipts.android.sync.model.SyncState

class DistanceLocationColumn(id: Int, syncState: SyncState, private val localizedContext: Context) :
    AbstractColumnImpl<Distance>(
        id,
        DistanceColumnDefinitions.ActualDefinition.LOCATION,
        syncState
    ) {

    override fun getValue(distance: Distance): String? {
        return distance.location
    }

    override fun getFooter(distances: List<Distance>): String {
        return localizedContext.getString(R.string.total)
    }
}
